idseq_dag/steps/run_lzw.py

```diff
@@ -51,8 +52,17 @@ def count_reads(self):
         self.should_count_reads = True
         self.counts_dict[self.name] = count.reads_in_group(self.output_files_local()[0:2])
 
+    # predict LZW score from sequence length based on model fit to mean LZW score across mean lengths
+    # model formula: lm(formula = y ~ log(x))
```

Comment:
Sorry, probably newbie question... what data (`x` and `y`) was used to get to these values? Did you just use all the data in our db? Want to make sure we can get back to them from the information here.

Reply:
I just read the explanation in the description better. Would still be good to clear state how to retrieve those values from the DB e.g. `select all lengths from DB for a particular set of hosts...`

Reply:
The x-values were `mean read length` for sequences from a sample, and the y-values were `mean lzw score` for the same sample. This was done on a subset of samples along a defined distribution.

Specifically, eight sequence length windows were defined and within each window, a maximum of ten samples were selected. For the lower and higher windows, data may not have been available to reach the ten samples. The distribution of selected samples were as follows: 

- 25 - 60bp (2 samples)
- 60 - 80bp (10 samples)
- 80 - 130 bp (10 samples)
- 130 - 180bp (10 samples)
- 180 - 220bp (10 samples)
- 220 - 280bp (10 samples)
- 280 - 500bp (10 samples)
- > 500 bp (1 sample)

Across a total of 63 samples, the average read length and LZW compression score were computed for the first 500 reads. From these values, a linear model was fit using the R stats package.

Reply:
Is there some of this info that we should include in the comments in the code?
This conversation might be lost after you merge.

Reply:
Added the following comment text:
Across a total of 63 samples with varying read lengths, the average read length and LZW compression score were computed for the first 500 reads. From these values, a linear model was fit using the R stats package.